cleanup.go

```diff
@@ -8,29 +8,76 @@ import (
 	"time"
 )
 
-// terminateOptions is a type that holds the options for terminating a container.
-type terminateOptions struct {
-	ctx     context.Context
-	timeout *time.Duration
-	volumes []string
+// DefaultTimeout for termination
+var DefaultTimeout = 10 * time.Second
+
+// TerminateOptions is a type that holds the options for terminating a container.
+type TerminateOptions struct {
+	ctx         context.Context
+	stopTimeout *time.Duration
+	volumes     []string
 }
 
 // TerminateOption is a type that represents an option for terminating a container.
-type TerminateOption func(*terminateOptions)
+type TerminateOption func(*TerminateOptions)
+
+// NewTerminateOptions returns a fully initialised TerminateOptions.
+// Defaults: StopTimeout: 10 seconds.
+func NewTerminateOptions(ctx context.Context, opts ...TerminateOption) *TerminateOptions {
+	options := &TerminateOptions{
+		stopTimeout: &DefaultTimeout,
```

Comment:
suggestion: make local to fix the race condition mentioned above.
```suggestion
	timeout := time.Second * 10
	options := &TerminateOptions{
		stopTimeout: &timeout,
```